stl/inc/xutility

```diff
@@ -2479,17 +2479,31 @@ namespace ranges {
     _NODISCARD constexpr auto _As_const_pointer(const _Ty* _Ptr) noexcept {
         return _Ptr;
     }
+
+    template <class _Ty>
+    using _Begin_adapted = decltype(_RANGES begin(_RANGES _Possibly_const_range(_STD declval<_Ty&>())));

```

Comment:
I don't like this name very much - `_Begin_adapted` does not say to me that this is about constifying stuff. I'd prefer something along the lines of `_Possibly_const_begin_t` or `_Begin_on_const` or smth. Same for `_End_adapted` and the rest.